core/client/verifier/pairs.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package verifier
```

Comment:
Is it necessary to place in a separate file? I usually find it quite confusing when a type implementation is distributed.

Reply:
It's not super necessary, but I do think of the functions in this file have a strict "depends on" relationship with `verifier.go`.  I might refactor these functions into a separate type.